app/src/controllers/log/utils.js

```diff
@@ -0,0 +1,46 @@
+import { PASSED, FAILED, MANY, NOT_FOUND } from 'common/constants/launchStatuses';
+
+const validForGrowthDuration = (item) =>
+  item.status === FAILED.toUpperCase() || item.status === PASSED.toUpperCase();
+
+export const calculateGrowthDuration = (historyItems) => {
+  const historyItemsLastIndex = historyItems.length - 1;
+  return historyItems
+    .reverse()
+    .map((item, index) => {
+      if (!validForGrowthDuration(item) || index >= historyItemsLastIndex) {
+        return item;
+      }
+
+      const newItem = { ...item };
+      let prevItemIndex = index + 1;
+      while (
+        !validForGrowthDuration(historyItems[prevItemIndex]) &&
+        prevItemIndex < historyItemsLastIndex
+      ) {
+        prevItemIndex += 1;
+      }
+      if (prevItemIndex <= historyItemsLastIndex) {
+        const prevDuration =
+          historyItems[prevItemIndex].end_time - historyItems[prevItemIndex].start_time;
+        const currentDuration = item.end_time - item.start_time;
+        const growth = currentDuration / prevDuration - 1;
+        growth > 0 && (newItem.growthDuration = `+${Math.round(growth * 100)}%`);
+      }
+      return newItem;
+    })
+    .reverse();
+};
+
+export const getHistoryItemProps = (filteredLaunchHistoryItem) => {
```

Comment:
Unclear name. What does this function do?

Reply:
This function is clarify history item's props according to it's status

Reply:
Fixed